compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1523,9 +1523,9 @@ object Types {
           if (mySig == null) mySig = Signature.NotAMethod
         case designator: Symbol =>
           uncheckedSetSym(designator)
-        case LocalName(underlying, space) =>
```

Comment:
Is the warning valid here?

Reply:
Yes

Reply:
Can you explain why? =)

Reply:
```scala
[warn] -- Warning: /Users/nicolasstucki/GitHub/dotty/compiler/src/dotty/tools/dotc/core/Types.scala:1543:22 
[warn] 1543 |        case LocalName(underlying, space) =>
[warn]      |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[warn]      |There is no best instantiation of pattern type dotty.tools.dotc.core.Designators.LocalName[dotty.tools.dotc.core.Names.Name]
[warn]      |that makes it a subtype of selector type dotty.tools.dotc.core.Designators.Designator.
[warn]      |Non-variant type variable N cannot be uniquely instantiated.
[warn]      |(This would be an error under strict mode)
```

We are pattern matching on a `Designator { type ThisType  <: Name }` but we try to match it against a `LocalName[N <: Name] { type ThisType = N }`. Note that the designator does not define the `N` type parameter. With these constraints we could instantiate `N = Typename` or `N = TermName`.